baseplate/lib/message_queue.py

```diff
@@ -103,22 +134,39 @@ def put(self, message: bytes, timeout: Optional[float] = None) -> None:
         raise TimedOutError
 
     def unlink(self) -> None:
-        """Remove the queue from the system.
-
-        The queue will not leave until the last active user closes it.
-
-        """
         self.queue.unlink()
 
     def close(self) -> None:
-        """Close the queue, freeing related resources.
+        self.queue.close()
 
-        This must be called explicitly if queues are created/destroyed on the
-        fly. It is not automatically called when the object is reclaimed by
-        Python.
 
-        """
-        self.queue.close()
+class InMemoryMessageQueue(MessageQueue):
```

Comment:
Jayme is right that we actually need two Queue classes for this.

In the baseplate service, we need the "RemoteQueue" to make a request to the sidecar container to place the item.

For this, the sidecar actually needs to expose a simple server with a put and get method. This could for example be a standard thrift server. You can also implement a simple binary protocol yourself.

In the sidecar, the server methods can use the `InMemoryMessageQueue` to store the items.